internal/cli/config/config.go

```diff
@@ -14,53 +14,117 @@ See the Mulan PSL v2 for more details.
 package config
 
 import (
-	"bytes"
-	"fmt"
+	"errors"
+	"io"
+	"net/http"
 
-	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
 
-	"github.com/oceanbase/ob-operator/internal/cli/generated/bindata"
 	"github.com/oceanbase/ob-operator/internal/cli/utils"
 )
 
-// component config for test
-var confPath = "internal/assets/cli-templates/component_config.yaml"
+// Chart struct to parse the chart.yaml
+type Chart struct {
+	apiVersion  string `yaml:"apiVersion"`
+	AppVersion  string `yaml:"appVersion"`
+	description string `yaml:"description"`
+	name        string `yaml:"name"`
+	chartType   string `yaml:"type"`
+	version     string `yaml:"version"`
+}
 
-func readComponentConf(path string) map[string]string {
-	components := make(map[string]string)
-	fileobj, err := bindata.Asset(path)
-	// panic if file not exists
+const (
+	stableVersion = "stable"
+	devVersion    = "master"
+)
+
+// ComponentList is the list of components that can be installed
+var ComponentList = []string{
+	"cert-manager", "ob-operator", "ob-dashboard", "local-path-provisioner", "local-path-provisioner-dev", "ob-operator-dev",
+}
+
+// ComponentUpdateList is the list of components that can be updated
+var ComponentUpdateList = []string{
+	"cert-manager", "ob-operator", "ob-dashboard", "local-path-provisioner",
+}
+
+var versionURLs = map[string]string{
+	"ob-dashboard":           "https://raw.githubusercontent.com/oceanbase/ob-operator/refs/heads/stable/charts/oceanbase-dashboard/Chart.yaml",
+	"local-path-provisioner": "https://raw.githubusercontent.com/rancher/local-path-provisioner/refs/tags/v0.0.30/deploy/chart/local-path-provisioner/Chart.yaml",
+}
+
+func getVersionFromChart(component string) (string, error) {
+	url, ok := versionURLs[component]
+	if !ok {
+		return "", errors.New("url not found for the component")
+	}
+
+	// get the yaml
+	resp, err := http.Get(url)
 	if err != nil {
-		panic(fmt.Errorf("Error reading component config file: %v", err))
+		return "", err
 	}
-	viper.SetConfigType("yaml")
-	err = viper.ReadConfig(bytes.NewBuffer(fileobj))
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.New("failed to fetch chart: " + resp.Status)
+	}
+
+	// parse the yaml
+	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		panic(fmt.Errorf("Read Config err:%v", err))
+		return "", err
 	}
-	if err := viper.UnmarshalKey("components", &components); err != nil {
-		panic(fmt.Errorf("Error decoding component config file: %v", err))
+	var chart Chart
+	err = yaml.Unmarshal(body, &chart)
+	if err != nil {
+		return "", err
+	}
+
+	return chart.AppVersion, nil
+}
+
+func getVersion(component string) (string, error) {
+	switch component {
+	case "ob-dashboard", "local-path-provisioner":
+		return getVersionFromChart(component)
+	case "cert-manager", "ob-operator":
+		return stableVersion, nil
+	case "ob-operator-dev", "local-path-provisioner-dev":
```

Comment:
I think `local-path-provisioner` has no `dev` version. The chart put in master branch of local-path-provisioner is always for production.

Reply:
Through the readme of `local-path-provisioner` repo it can be seen that they provide `dev` and `stable` version. Do we need to provide only the version under the master branch here?

Reply:
As far as I see, the chart in the master branch is always of the stable version. The `stable` and `dev` are used to describe the status of `deploy/local-path-storage.yaml`, I thought.

Reply:
If users want to install `local-path-provisioner-dev`, the okctl will install it through `deploy/local-path-storage.yaml` right?

Reply:
Yes, for okctl: 
- `stable`->`https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.30/deploy/local-path-storage.yaml`
- `dev`-> `https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml`
And for now, `getVersion` function is used to get v0.0.30 for stable version.
Actually, I just found that we can just set stable version to `v0.0.30` temporarily, no need to visit the url and parse the version.

Reply:
It's not suggested to hardcode the version in code.

Reply:
Removing local-path-provisioner-dev is a more reasonable choice. Because we are not responsible for what it would happen if users install local-path-provisioner-dev